worldedit-core/src/main/java/com/sk89q/worldedit/extent/transform/BlockTransformExtent.java

```diff
@@ -228,27 +228,28 @@ public static <B extends BlockStateHolder<B>> B transform(B block, Transform tra
                     }
 
                     String value = (String) result.getState(property);
+                    if (isRailShape(value)) {
```

Comment:
Just as an optimisation, it might make sense to check that the `property` here has the correct number of values too, before checking that the actual values match